### PR TITLE
fix storage policy prompt display error

### PR DESCRIPTION
**Pull request by xiaolei565**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
correcting the message of modify storage policy https://github.com/ClickHouse/ClickHouse/issues/51516
### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Disks/StoragePolicy.cpp [hunk 1]
```diff
@@ -302,15 +302,15 @@ void StoragePolicy::checkCompatibleWith(const StoragePolicyPtr & new_storage_pol
     for (const auto & volume : getVolumes())
     {
         if (!new_volume_names.contains(volume->getName()))
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "New storage policy {} shall contain volumes of old one", backQuote(name));
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "New storage policy {} shall contain volumes of old one", backQuote(new_storage_policy->getName()));
```
Comment (alexey-milovidov) [inline src/Disks/StoragePolicy.cpp:305]: Let's print both names, new and old.

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit ffa0f8d75bc249e6a133fe3fa95b8385ee16ebef with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51519/ffa0f8d75bc249e6a133fe3fa95b8385ee16ebef/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Disks/StoragePolicy.cpp (hunk 1 above)
Comment (xiaolei565) [reply src/Disks/StoragePolicy.cpp]: good idea, I've got it

**Review by alexey-milovidov — APPROVED** [review]

**Comment by xiaolei565** [comment]
@alexey-milovidov 

hello，What should I do when encountering CI reporting the following error？

```
@robot-clickhouse-ci-1
Mergeable Check — failed: Stateless tests (release, DatabaseReplicated) [3/4], Stateless tests (ubsan) [2/2]
Required
[Details](https://github.com/ClickHouse/ClickHouse/actions/runs/5406516337)
@robot-clickhouse-ci-1
Stateless tests (release, DatabaseReplicated) [3/4] — fail: 1, passed: 1281, skipped: 34
```


02442_auxiliary_zookeeper_endpoint_id | FAIL | 0.43
-- | -- | --
2023-06-29 03:29:20 [66d03adae516] 2023.06.29 03:29:20.738122 [ 11182 ] {a53bce73-6f3b-443a-9dab-8d3e1657f71c} <Error> executeQuery: Code: 999. Coordination::Exception: ZooKeeper root doesn't exist. You should create root node /test/chroot/auxiliary_zookeeper2 before start. (No node). (KEEPER_EXCEPTION) (version 23.6.1.1363) (from 0.0.0.0:0) (comment: 02442_auxiliary_zookeeper_endpoint_id.sql) (in query: /* ddl_entry=query-0000000006 */ CREATE TABLE test_gvyyfrir.t2 UUID 'f250a35a-1a38-475d-84e3-668efffb5853' (`x` Int32) ENGINE = ReplicatedMergeTree('zookeeper2:/test/02442/{database}/t/auto_{shard}', 'r1auto_{replica}') ORDER BY x SETTINGS index_granularity = 43752, min_bytes_for_wide_part = 0, ratio_of_defaults_for_sparse_serialization = 1., merge_max_block_size = 13959, prefer_fetch_merged_part_size_threshold = 10737418240, vertical_merge_algorithm_min_rows_to_activate = 1000000, vertical_merge_algorithm_min_columns_to_activate = 1, min_merge_bytes_to_use_direct_io = 10737418240, index_granularity_bytes = 787014, allow_vertical_merges_from_compact_to_wide_parts = 1, marks_compress_block_size = 94195, primary_key_compress_block_size = 93693), Stack trace (when copying this message, always include the lines below): 2023-06-29 03:29:20  2023-06-29 03:29:20 0. ./build_docker/./src/Common/Exception.cpp:92: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000e1ec155 in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 1. ./build_docker/./contrib/llvm-project/libcxx/include/string:1499: Coordination::Exception::Exception(String const&, Coordination::Error, int) @ 0x00000000151ff151 in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 2. ./build_docker/./contrib/llvm-project/libcxx/include/string:1499: Coordination::Exception::Exception(String const&, Coordination::Error) @ 0x00000000151ff47f in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 3. ./build_docker/./src/Common/ZooKeeper/ZooKeeper.cpp:0: zkutil::ZooKeeper::init(zkutil::ZooKeeperArgs) @ 0x0000000015203d73 in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 4. ./build_docker/./src/Common/ZooKeeper/ZooKeeper.cpp:0: zkutil::ZooKeeper::ZooKeeper(Poco::Util::AbstractConfiguration const&, String const&, std::shared_ptr<DB::ZooKeeperLog>) @ 0x0000000015205d23 in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 5. ./build_docker/./src/Interpreters/Context.cpp:2816: DB::Context::getAuxiliaryZooKeeper(String const&) const @ 0x0000000012f50b04 in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 6. ./build_docker/./src/Storages/StorageReplicatedMergeTree.cpp:0: DB::StorageReplicatedMergeTree::StorageReplicatedMergeTree(String const&, String const&, bool, DB::StorageID const&, String const&, DB::StorageInMemoryMetadata const&, std::shared_ptr<DB::Context>, String const&, DB::MergeTreeData::MergingParams const&, std::unique_ptr<DB::MergeTreeSettings, std::default_delete<DB::MergeTreeSettings>>, bool, DB::RenamingRestrictions, bool) @ 0x0000000014272061 in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 7. ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:1460: DB::create(DB::StorageFactory::Arguments const&) @ 0x0000000014a8ed78 in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 8. ./build_docker/./src/Storages/StorageFactory.cpp:225: DB::StorageFactory::get(DB::ASTCreateQuery const&, String const&, std::shared_ptr<DB::Context>, std::shared_ptr<DB::Context>, DB::ColumnsDescription const&, DB::ConstraintsDescription const&, bool) const @ 0x000000001419eebb in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 9. ./build_docker/./contrib/llvm-project/libcxx/include/__utility/swap.h:37: DB::InterpreterCreateQuery::doCreateTable(DB::ASTCreateQuery&, DB::InterpreterCreateQuery::TableProperties const&, std::unique_ptr<DB::DDLGuard, std::default_delete<DB::DDLGuard>>&) @ 0x0000000013863d4c in /usr/lib/debug/usr/bin/clickhouse.debug 2023-06-29 03:29:20 10. ./build_docker/./src/Interpreters/InterpreterCreateQuery.cpp:0: DB::InterpreterCrea

**Comment by alexey-milovidov** [comment]
Wait while it is fixed, or try to figure it out on your own.

**Comment by xiaolei565** [comment]
> Wait while it is fixed, or try to figure it out on your own.

OK，thanks for your reply

**Comment by xiaolei565** [comment]
@alexey-milovidov @evillique  Hello，Please help me merge this PR，thanks.

**Pull request merged**